tso-server/src/main/java/org/apache/omid/tso/Batch.java

```diff
@@ -51,41 +51,45 @@
 
     }
 
-    boolean isFull() {
-
-        Preconditions.checkState(numEvents <= size, "numEvents > size");
-        return numEvents == size;
+    PersistEvent get(int idx) {
+        Preconditions.checkState(numEvents > 0 && 0 <= idx && idx < numEvents);
```

Comment:
nitpicking: better to have error message with human readable text there otherwise it throws IllegalArgumentException without any explanation